tests/test_ess_csc.py

```diff
@@ -458,3 +458,70 @@ def topic_callback(self, data: salobj.BaseMsgType, attr_name: str) -> None:
             else:
                 sensor_data[data.timestamp] = data
         self.data_event.set()
+
+    async def test_rpi_data_client_loses_single_connecton(self) -> None:
+        """The CSC should fault when the RPiDataClient loses its connection
+        to the server and the RpiDataClient should reconnect when the CSC is
+        set to ENABLED again.
+        """
+        # Start the MockServer for manual control.
+        await self.start_mock_server()
+        async with self.make_csc(
+            initial_state=salobj.State.STANDBY,
+            config_dir=TEST_CONFIG_DIR,
+            simulation_mode=0,
+            override="test_one_temp_sensor.yaml",
+        ):
+            await self.assert_next_summary_state(salobj.State.STANDBY)
```

Comment:
Why all these step-wise state transitions? It would be simpler to specify initial_state=salobj.State.ENABLED in the call to make_csc and assert that the next summary state is ENABLED, then go from there.

Reply:
Yes you're right. I was going through each state transition manually to investigate which transition introduced the error. I will change that.